clip2onnx/model_card.py

```diff
@@ -0,0 +1,165 @@
+    """
+    Merged ONNX Model Name.
+    If you decide to export the model on the huggingface, 
+    please customize the onnx_visual and onnx_textual fields.
+    """
+
+    onnx_visual_path: Path = field(default_factory=Path)
+    onnx_textual_path: Path = field(default_factory=Path)
+    model_dir: Path = Path("model")
+    """
+    Default export path:
+      - visual: [project_dir]/model/*[self.model_name]/[VERSION]/[self.onnx_visual]
+      - textual: [project_dir]/model/*[self.model_name]/[VERSION]/[self.onnx_textual]
+    """
+
+    DEFAULT_TEXTUAL_FIELDS = {
+        "export_params": True,
+        "input_names": ["TEXT"],
+        "output_names": ["TEXT_EMBEDDING"],
+        "dynamic_axes": {"TEXT": {0: "text_batch_size"}, "TEXT_EMBEDDING": {0: "text_batch_size"}},
+    }
+    """
+    Template parameter of the textual-part appended to `torch.onnx.export`
+    """
+
+    DEFAULT_VISUAL_FIELDS = {
+        "export_params": True,
+        "input_names": ["IMAGE"],
+        "output_names": ["IMAGE_EMBEDDING"],
+        "dynamic_axes": {
+            "IMAGE": {0: "image_batch_size"},
+            "IMAGE_EMBEDDING": {0: "image_batch_size"},
+        },
+    }
+    """
+    Template parameter of the visual-part appended to `torch.onnx.export`
+    """
+
+    def __post_init__(self):
+        _prefix = self.model_name
+
+        if not self.tag and (not self.onnx_visual or not self.onnx_textual):
+            logging.warning(
+                "If you decide to export the model on the huggingface, "
+                "please customize the onnx_visual and onnx_textual fields."
+            )
+            _prefix = self.model_name.split(":")[-1]
+
+        # fixme: more standardized model naming
+        if not self.tag:
+            inner_ = _prefix.split("/")[-1]
+            self.onnx_visual = f"visual_CLIP_{inner_}.onnx"
+            self.onnx_textual = f"textual_CLIP_{inner_}.onnx"
+        else:
+            self.onnx_visual = f"visual_CLIP_{_prefix}.{self.tag}.onnx"
+            self.onnx_textual = f"textual_CLIP_{_prefix}.{self.tag}.onnx"
+
+        _suffix = 1
+        for _ in range(100):
+            pre_dir = self.model_dir.joinpath(_prefix, f"v{_suffix}")
+            if not pre_dir.exists():
+                pre_dir.mkdir(parents=True, exist_ok=True)
+                break
+            _suffix += 1
+
+        self.onnx_visual_path = self.model_dir.joinpath(_prefix, f"v{_suffix}", self.onnx_visual)
+        self.onnx_textual_path = self.model_dir.joinpath(_prefix, f"v{_suffix}", self.onnx_textual)
+        logging.info(self.onnx_textual_path)
+        logging.info(self.onnx_textual_path)
+
+    @classmethod
+    def from_template(cls, template):
+        return cls(
+            **{
+                key: (template[key] if val.default == val.empty else template.get(key, val.default))
+                for key, val in inspect.signature(cls).parameters.items()
+            }
+        )
+
+    def __call__(self, *args, **kwargs):
```

Comment:
Let's be explicit and simply call this method `export_to_onnx` instead of overriding `__call__`